pkg/groupsync/groups.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupsync
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+// ReadGroupClient provides read operations for a group system.
+type ReadGroupClient interface {
```

Comment:
nit: `GroupReader` and `GroupWriter` and `GroupReadWriter` seems simpler names. This is pretty common in Go stdlib

Reply:
done.